mlrun/api/db/sqldb/session.py

```diff
@@ -60,6 +60,7 @@ def _init_engine(dsn=None):
         kwargs = {
             "pool_size": pool_size,
             "max_overflow": max_overflow,
+            "isolation_level": config.db.isolation_level,
```

Comment:
why doing it globally and not per-query?
controlling the isolation level is very delicate, moving from one isolation to other should be taken very carefully. and thus, should be pretty static.
that being said, if specific query / function needs different isolation, for any reason, that should be handy. other than that, let it be the default

Reply:
leaving the old behavior will just leave us open to more bugs related data consistency, I don't see why to stay with the old behavior if it results inconsistencies with the data.

I made it configurable as we still want to be able to respond if some issues will occur because of that isolation level and just rollback to the old one.

Reply:
I am not saying to revert, im asking why not allowing the isolation level to be configurable in 2 ways
1. per-query
2. globally.

if per-query is given, use it, otherwise default to globally (which is `config.db.isolation_level`)

Reply:
If we take our current flow with creating project, if we make it configurable per-query we can't define it at the endpoint level, because we don't know in which flow we are, meaning we will need to do it kinda in the same area as now and create new session with specific isolation level, which makes the current `.commit()` redundant but also makes the isolation level definition redundant because we already created new session which means its transaction cache is clear. Now I do get that in that flow, even if user define some other isolation level we would like that specific flow to have `READ COMMITTED` isolation mode to avoid the bug.

Option 1 : To create `READ COMMITTED` session on the create project endpoint in regards the flow.
Option 2: To create `READ COMMITTED` session when resolving in which flow we are.
Option 3: Leave the `.commit()` in that flow.